File: src/main/java/com/telegram/folobot/dto/FoloUserDto.java
package com.telegram.folobot.dto;

import lombok.Data;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.Objects;
import java.util.stream.Stream;

@Data
@Accessors(chain = true)
@RequiredArgsConstructor
public class FoloUserDto implements Serializable {
    private Long userId;
    private Long mainId;
    private String name;
    private String tag;

    public FoloUserDto(Long userId) {
        this(userId, "");
    }

    public FoloUserDto(Long userId, String tag) { this(userId, null, tag); }

    public FoloUserDto(Long userId, Long mainId, String tag) {
        this.userId = userId;
        this.mainId = mainId;
        this.tag = tag;
    }

    public Long getMainUserId() {
        return Stream.of(mainId, userId)
                .filter(f -> !Objects.isNull(f) && f != 0L)
                .findFirst()
                .orElse(null);
    }

    public Long getMainId() { return !Objects.isNull(mainId) ? mainId : 0L; }

    public String getName() { return !Objects.isNull(tag) && !tag.isEmpty() ? tag : name != null ? name : ""; }

    public String getRealName() { return !Objects.isNull(name) ? name : ""; }

    public String getTag() { return !Objects.isNull(tag) ? tag : ""; }
}

File: src/main/kotlin/com/telegram/folobot/dto/FoloUserDto.kt
package com.telegram.folobot.dto

import java.io.Serializable

class FoloUserDto(
    val userId: Long,
    var mainId: Long = 0L,
    var anchor: Boolean = false,
    var name: String = "",
    var tag: String = ""
) : Serializable {

    /**
     * Получить основоного пользователя
     */
    fun getMainUserId() : Long {
        return if (mainId != 0L) mainId else userId
    }

    /**
     * Получить тэг, если он путст имя
     */
    fun getTagName(): String { return if (tag.isNotEmpty()) tag else name }

    /**
     * Установить основного пользователя и вернуть себя
     * @return [FoloUserDto]
     */
    fun setMainId(mainId: Long): FoloUserDto {
        this.mainId = mainId
        return this
    }

    /**
     * Установить якорь и вернуть себя
     * @return [FoloUserDto]
     */
    fun setAnchor(anchor: Boolean): FoloUserDto {
        this.anchor = anchor
        return this
    }

    /**
     * Установить имя и вернуть себя
     * @return [FoloUserDto]
     */
    fun setName(name: String?): FoloUserDto {
        if (name != null) {
            this.name = name
        }
        return this
    }

    /**
     * Установить тэг и вернуть себя
     * @return [FoloUserDto]
     */
    fun setTag(tag: String): FoloUserDto {
        this.tag = tag
        return this
    }
}